x/staking/keeper/validator.go

```diff
@@ -53,6 +53,28 @@ func (k Keeper) mustGetValidatorByConsAddr(ctx sdk.Context, consAddr sdk.ConsAdd
 	return validator
 }
 
+func (k Keeper) GetValidatorByUnbondingOpId(
+	ctx sdk.Context, id uint64,
+) (val types.Validator, found bool) {
+	store := ctx.KVStore(k.storeKey)
+	indexKey := types.GetUnbondingOpIndexKey(id)
```

Comment:
The GetValidatorByUnbondingOpId etc functions will be less useful and general if they take an indexKey, since indexKeys are an internal abstraction of this system. That's why I have them taking an id. A good solution is probably to make them take an indexKey and also make them private.

Reply:
But this needs to be weighed against how much lookups actually cost. I assume they are very cheap, but I guess I don't know for certain.

Reply:
Actually, `GetUnbondingOpIndexKey` is a simple byte concatenation, not a lookup. So it is basically free, and may even be optimized into code that does what you suggest. So i'm going to leave the code as is, since id is clearer and easier to work with.

Reply:
Yeah, that's true. Sorry for the comment.